pytest_bdd/scenario.py

```diff
@@ -298,9 +277,9 @@ def scenario(feature_name, scenario_name, encoding="utf-8", example_converters=N
     caller_function = caller_function or get_caller_function()
 
     # Get the feature
-    base_path = get_fixture(caller_module, "pytestbdd_feature_base_dir")
-    strict_gherkin = get_fixture(caller_module, "pytestbdd_strict_gherkin")
-    feature = Feature.get_feature(base_path, feature_name, encoding=encoding, strict_gherkin=strict_gherkin)
+    base_dir = get_base_dir(caller_module)
```

Comment:
With the previous implementation, we could have test modules that don't want to define a base_dir (i.e. defaulting to the directory of the test module), and other tests modules that want to have feature files in a different location.
With this implementation, the use case is no longer possible.
What we could do is to add a `base_dir` parameter to (which defaults to None) to the `scenario` and `scenarios` functions, so that we still have a way to solve this use case.

Reply:
I am going to try to implement your solution.

Do you know any valid use case for that additional feature? Or by having that feature we encourage anti-patterns here? ;-)

However I agree that we shouldn't silently remove features. If we want to remove it, we should deprecate it firstly as someone may use it.

Reply:
I don't know how people use this library, so I am just careful to not kill use cases. We can even try a lazy approach, and implement what I was suggesting only if somebody opens an issue.
But in my case, I really need to be able to specify `strict_gherking` on some specific scenarios (by having it as a parameter of `scenario{,s}`). So if we implement this, let's also implement base_dir, it won't be much more difficult.

Reply:
Done